src/test/java/codeu/controller/TestDataGeneratorTest.java

```diff
@@ -0,0 +1,168 @@
+package codeu.controller;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.persistence.PersistentDataStoreException;
+import codeu.model.store.persistence.PersistentStorageAgent;
+
+public class TestDataGeneratorTest {
+
+  private PersistentStorageAgent mockPersistentStorageAgent;
+  private TestDataGenerator mockTestDataGenerator;
+
+  private UserStore mockUserStore;
+  private ConversationStore mockConversationStore;
+  private MessageStore mockMessageStore;
+
+  @Before
+  public void setup() throws PersistentDataStoreException {
+
+    mockPersistentStorageAgent = Mockito.mock(PersistentStorageAgent.class);
+    Mockito.when(mockPersistentStorageAgent.loadUsers()).thenReturn(new ArrayList<User>());
+    Mockito.when(mockPersistentStorageAgent.loadConversations()).thenReturn(new ArrayList<Conversation>());
+    Mockito.when(mockPersistentStorageAgent.loadMessages()).thenReturn(new ArrayList<Message>());
+
+    mockTestDataGenerator = TestDataGenerator.getTestInstance(mockPersistentStorageAgent);
+
+    setUpStores();
+  }
+
+  void setUpStores() {
+
+    mockUserStore = UserStore.getTestInstance(mockPersistentStorageAgent);
+    mockTestDataGenerator.setUserStore(mockUserStore);
+
+    mockConversationStore = ConversationStore.getTestInstance(mockPersistentStorageAgent);
+    mockTestDataGenerator.setConversationStore(mockConversationStore);
+
+    mockMessageStore = MessageStore.getTestInstance(mockPersistentStorageAgent);
+    mockTestDataGenerator.setMessageStore(mockMessageStore);
+
+  }
+
+  @Test
+  public void testAddTestUsers() {
+
+    int numUsersBefore = mockUserStore.getAllUsers().size();
```

Comment:
The tests should all run independently from a known starting state, so you should be able to replace all of these "before" numbers with 0s (and inline them as appropriate).

If you'd like to also test that adding test entities can be done multiple times or when there's already data in the DB, that can be tested explicitly with multiple calls or setting a new return value for mockPersistentStorageAgent.